src/scipp/core/bins.py

```diff
@@ -124,12 +125,20 @@ def _data(self):
             return self._obj
 
     def __mul__(self, lut: lookup):
-        copy = self._obj.copy()
+        target_dtype = (
+            scalar(1, dtype=self.constituents['data'].dtype)
```

Comment:
I feel we should add `self.dtype`, similar to `self.unit`?

Can you check how expensive `constituents` is? I think it used to unzip the indices (allocating memory), but now it probably just returns 2 stride-2 variables for `begin` and `end`, i.e., is cheap?

Reply:
Well, `self.unit` is implemented in terms of `constituents`. So unless we implement `dtype` in `C++`, we'd still pay the cost.
But I agree that it would be nicer to have a dtype property on `Bins`.

Reply:
And yes, it does unzip the indices.

Reply:
I don't know what you want to say with this. But here is the implementation: https://github.com/scipp/scipp/blob/92717a5c470030b87cae5c7db38f059bcd32aedb/lib/python/bins.cpp#L84

Reply:
Probably `unzip` does this under the hood, i.e., it does not cause copies?

Reply:
It looks like it doesn't make a copy: (this messed up the binned variable, but hey ho)
```python
import scipp as sc

da = sc.data.binned_x(10, 3)
print(da.bins.constituents['begin'])
da.bins.constituents['begin'][0] = 1
print(da.bins.constituents['begin'])
```
->
```
<scipp.Variable> (x: 3)      int64        <no unit>  [0, 7, 7]
<scipp.Variable> (x: 3)      int64        <no unit>  [1, 7, 7]
```

Reply:
👍 ... then this is *reasonably* fast, i.e., usable for `unit` and `dtype`.